src/spawn.rs

```diff
@@ -0,0 +1,463 @@
+        }
+
+        Ok(self)
+    }
+
+    /// Add a [dup2](https://pubs.opengroup.org/onlinepubs/9699919799/functions/dup2.html) action. See
+    /// [posix_spawn_file_actions_adddup2](https://pubs.opengroup.org/onlinepubs/9699919799/functions/posix_spawn_file_actions_adddup2.html).
+    #[doc(alias("posix_spawn_file_actions_adddup2"))]
+    pub fn add_dup2<Fd1: AsFd, Fd2: AsFd>(
+        &mut self,
+        fd: Fd1,
+        newfd: Fd2,
+    ) -> Result<()> {
+        let res = unsafe {
+            libc::posix_spawn_file_actions_adddup2(
+                &mut self.fa as *mut libc::posix_spawn_file_actions_t,
+                fd.as_fd().as_raw_fd(),
+                newfd.as_fd().as_raw_fd(),
+            )
+        };
+        if res != 0 {
+            return Err(Errno::from_raw(res));
+        }
+
+        Ok(())
+    }
+
+    feature! {
+    #![all(feature = "fs", feature = "term")]
+    /// Add an open action. See
+    /// [posix_spawn_file_actions_addopen](https://pubs.opengroup.org/onlinepubs/9699919799/functions/posix_spawn_file_actions_addopen.html).
+    #[doc(alias("posix_spawn_file_actions_addopen"))]
+    pub fn add_open<Fd: AsFd, P: ?Sized + NixPath>(
+        &mut self,
+        fd: Fd,
+        path: &P,
+        oflag: OFlag,
+        mode: Mode,
+    ) -> Result<()> {
+        let res = path.with_nix_path(|cstr| unsafe {
+            libc::posix_spawn_file_actions_addopen(
+                &mut self.fa as *mut libc::posix_spawn_file_actions_t,
+                fd.as_fd().as_raw_fd(),
+                cstr.as_ptr(),
+                oflag.bits(),
+                mode.bits(),
+            )
+        })?;
+        if res != 0 {
+            return Err(Errno::from_raw(res));
+        }
+
+        Ok(())
+    }
+    }
+
+    /// Add a close action. See
+    /// [posix_spawn_file_actions_addclose](https://pubs.opengroup.org/onlinepubs/9699919799/functions/posix_spawn_file_actions_addclose.html).
+    #[doc(alias("posix_spawn_file_actions_addclose"))]
+    pub fn add_close<Fd: AsFd>(&mut self, fd: Fd) -> Result<()> {
+        let res = unsafe {
+            libc::posix_spawn_file_actions_addclose(
+                &mut self.fa as *mut libc::posix_spawn_file_actions_t,
+                fd.as_fd().as_raw_fd(),
+            )
+        };
+        if res != 0 {
+            return Err(Errno::from_raw(res));
+        }
+
+        Ok(())
+    }
+}
+
+impl Drop for PosixSpawnFileActions {
+    fn drop(&mut self) {
+        unsafe {
+            libc::posix_spawn_file_actions_destroy(
+                &mut self.fa as *mut libc::posix_spawn_file_actions_t,
+            );
+        }
+    }
+}
+
+// Specifically for use with posix_spawn and posix_spawnp.
+// https://github.com/rust-lang/libc/issues/1272
+unsafe fn to_exec_array<S: AsRef<CStr>>(args: &[S]) -> Vec<*mut libc::c_char> {
+    let mut v: Vec<*mut libc::c_char> = args
+        .iter()
+        .map(|s| s.as_ref().as_ptr() as *mut libc::c_char)
```

Comment:
For pointer casting, we prefer `.cast()` and `.cast_mut()`:

https://github.com/nix-rust/nix/blob/master/CONVENTIONS.md#pointer-type-casting

```suggestion
        .map(|s| s.as_ref().as_ptr().cast_mut())
```